syntax = "proto3";

package NAlice.NNlu;

option go_package = "a.yandex-team.ru/alice/nlu/proto/dataset_info";
option java_package = "ru.yandex.alice.nlu.proto.dataset_info";


import "alice/protos/data/language/language.proto";


message TSlotMarkupInfo {
    optional string Description = 1 [json_name = "description"];
    optional string Guideline = 2 [json_name = "guideline"];
    string Table = 3 [json_name = "table"];
    optional string Ticket = 4 [json_name = "ticket"];
    repeated string Slots = 5 [json_name = "slots"];
}

message TVoiceRecordingsInfo {
    optional string Description = 1 [json_name = "description"];
    string Table = 2 [json_name = "table"];
    optional string Ticket = 3 [json_name = "ticket"];
}

message TDatasetInfo {
    optional string Description = 1 [json_name = "description"];
    optional string Guideline = 2 [json_name = "guideline"];

    ELang Language = 3 [json_name = "language"];
    EDialect Dialect = 4 [json_name = "dialect"];

    optional string AcceptDataTable = 5 [json_name = "accept_data_table"];
    optional string DevDataTable = 6 [json_name = "dev_data_table"];
    optional string KpiDataTable = 7 [json_name = "kpi_data_table"];

    optional string Ticket = 8 [json_name = "ticket"];

    enum EOrigin {
        UNKNOWN = 0;
        HANDCRAFTED = 1;
        TRANSLATED_SYNTETIC = 2;
    }

    EOrigin Origin = 9 [json_name = "origin"];

    optional TSlotMarkupInfo SlotMarkupInfo = 10 [json_name = "slot_markup_info"];
    optional TVoiceRecordingsInfo VoiceRecordingsInfo = 11 [json_name = "voice_recordings_info"];

    bool Deprecated = 12 [json_name = "deprecated"];
}

message TDatasetInfoList {
    repeated TDatasetInfo DatasetInfos = 1 [json_name = "dataset_infos"];
}

message TIntentDatasets {
    map<string, TDatasetInfoList> IntentDatasets = 1 [json_name = "intent_datasets"];
}
